index.js

```diff
@@ -1,5 +1,100 @@
 const battleship = () => {
-  return 'The winner is...?'
+  let player1 = {
+    name: 'player1',  
+    shipCount: 0,
+    gameBoard: []
+    } 
+  
+  let player2 = {
+    name: 'player2',  
+    shipCount: 0,
+    gameBoard: []
+    } 
+
+  player1.name = prompt('player1Name');
+  player2.name = prompt('player2Name');
+
+  let boardSize = 4;
+  let startShipCount = 4;  
+ 
+  for(let i = 0 ; i < boardSize; i++){
+    let boardRow = []
+    for (let j = 0; j < boardSize; j++){
+      boardRow.push(0)
+    } 
+    player1.gameBoard.push([].concat(boardRow));
+    player2.gameBoard.push([].concat(boardRow)); 
+  }
+ 
+  function placeShip(player) {
+    let x = Math.floor(Math.random() *4)
+    let y = Math.floor(Math.random() *4)
+    if (player.gameBoard[x][y] === 1) {
+        return
+    } else {
+      console.log(x,y);
+      
+      player.gameBoard[x][y]= 1;
+      player.shipCount++;
+    } 
+  }
+  
+  while(player1.shipCount < startShipCount) {   
+      placeShip(player1);
+    }  
+    
+  while(player2.shipCount < startShipCount){
+      placeShip(player2);
+    }  
+    function showBoard(player) {
+      for(let i=0; i<player.gameBoard.length; i++){
+      //console.log(gameBoard[i].toString());
+    }
+  }
+    let currentPlayer = player1;
+    while(player1.shipCount>0 && player2.shipCount>0){
+    
+    console.log('Current Player', currentPlayer);
+      
+    //player1.shipCount--;
+
+    
+  console.log(player1.gameBoard);
+  console.log(player2.gameBoard);
+  
+  let opponent;
+    if (currentPlayer === player1){
+     opponent = player2;
+  } else {
+     opponent = player1;
+  }
+  let x = prompt('X');
```

Comment:
Current player name can be displayed in the prompt to make it clear which players turn